docs: Document what should appear in migration guide (#2138)
faker-js/faker: CONTRIBUTING.md

}
```

## Documenting changes for new major versions

> Each major version has an upgrading guide eg https://next.fakerjs.dev/guide/upgrading.html

Reviewer: ```suggestion
Each major version has an upgrading guide e.g. https://next.fakerjs.dev/guide/upgrading.html
```

> Each major version has an upgrading guide e.g. https://next.fakerjs.dev/guide/upgrading.html

Reviewer 2: ```suggestion
Each major version has an upgrading guide, e.g. https://next.fakerjs.dev/guide/upgrading.html.
```

> Each major version has an upgrading guide, e.g. https://next.fakerjs.dev/guide/upgrading.html.

Reviewer 3: ```suggestion
Each major version has an upgrading guide, e.g. [fakerjs.dev/guide/upgrading](https://next.fakerjs.dev/guide/upgrading.html).
```

> Each major version has an upgrading guide, e.g. [fakerjs.dev/guide/upgrading](https://next.fakerjs.dev/guide/upgrading.html).

Reviewer: IMO the text should at least refer to the same domain as the actual target.

```suggestion
Each major version has an upgrading guide, e.g. [next.fakerjs.dev/guide/upgrading](https://next.fakerjs.dev/guide/upgrading.html).
```

> While developing new features and bugfixes for the new release, changes are added to the migration guide to aid developers when the version is released.

Reviewer 2: ```suggestion
While developing new features and fixing bugs for a new release, changes are added to the migration guide to aid developers when the version is released.
```

The general principle is to document anything which requires a normal user of the library to change their code which uses Faker when upgrading to the new major version.

> There are two sections

Reviewer: ```suggestion
There are two sections:
```

- Breaking changes (user MUST change their code)

> - Deprecations and other changes (user SHOULD change their code but it will still work for this major version if they don't)

Reviewer 2: ```suggestion
- Deprecations and other changes (user SHOULD change their code but it will still work for this major version even if they don't)
```

Not every change needs to be in the migration guide. If it is too long, it becomes hard for users to spot the important changes.

### Should be in the guide

> - Breaking changes, eg removal of methods
> - Behavior changes, eg a different default for a parameter, or a parameter becoming required
> - Whole modules renaming (eg faker.name to faker.person)
> - Locale renames
> - Changes to minimum versions eg requiring a new version of Node
> - Changes to how Faker is imported

Reviewer 2: ```suggestion
- Breaking changes, e.g. removal of methods
- Behavior changes, e.g. a different default for a parameter, or a parameter becoming required
- Whole modules renaming (e.g. faker.name to faker.person)
- Locale renames
- Changes to minimum versions e.g. requiring a new version of Node
- Changes to how Faker is imported
```

### Doesn't need to be in the guide

- New locales
- Changes to locale data in existing locales
- Bugfixes where it's unlikely anyone was relying on the old behavior (eg broken values in locale files)
- New methods and parameters

> - Straightforward method aliases, eg where a method or parameter is renamed but the old name still works identically. (Runtime warnings will already guide the user in this case)

Reviewer 2: ```suggestion
- Straightforward method aliases, e.g. where a method or parameter is renamed but the old name still works identically. (Runtime warnings will already guide the user in this case)
```

> - Changes to locale definition files which only affect usage via `faker.fake` e.g. if a definition file is renamed, but the public API for the method stays the same

Reviewer 3: isn't it `faker.helpers.fake`?

> - Changes to locale definition files which only affect usage via `faker.helpers.fake` e.g. if a definition file is renamed, but the public API for the method stays the same

Reviewer 2: ```suggestion
- Changes to locale definition files which only affect usage via `faker.helpers.fake`, e.g. if a definition file is renamed, but the public API for the method stays the same
```
